logstash-core/lib/logstash/pipeline.rb

```diff
@@ -200,9 +211,57 @@ def filters?
     return @filters.any?
   end
 
+  def start
+    # Since we start lets assume that the metric namespace is cleared
+    # this is useful in the context of pipeline reloading
+
+    clear_pipeline_metrics
+    collect_stats
+
+    logger.debug("Starting pipeline", :pipeline_id => pipeline_id)
+
+    @finished_execution = Concurrent::AtomicBoolean.new(false)
+
+    @thread = Thread.new do
+      begin
+        LogStash::Util.set_thread_name("pipeline.#{pipeline_id}")
+        run
+        @finished_execution.make_true
+      rescue => e
+        close
+        logger.error("Pipeline aborted due to error", :exception => e, :backtrace => e.backtrace)
```

Comment:
I've revisited all the logging statement int **pipeline.rb** and I have added a `#default_logging_keys` methods making it easy to add new values to the logging statements.